pkgs/development/libraries/libspng/default.nix

```diff
@@ -0,0 +1,23 @@
+{ lib, fetchFromGitHub, stdenv, zlib, ninja, meson }:
+
+stdenv.mkDerivation rec {
+  pname = "libspng";
+  version = "0.7.0-rc3";
+
+  src = fetchFromGitHub {
+    owner = "randy408";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0n91mr06sr34cqq91738251iaw21h5c4jgjpn0kqfx69ywxcl9fj";
+  };
+
+  buildInputs = [ zlib ];
+  nativeBuildInputs = [ ninja meson ];
+
```

Comment:
I think you need different build steps here https://libspng.org/docs/build/
The default build step is running `make`. You'll probably need to include your custom buildPhase

When you build it, can you ls into ./result and give the content of the output ? just to make sure it build correctly.

Reply:
```
result
├── include
│  └── spng.h
└── lib
   ├── libspng.so -> libspng.so.0
   ├── libspng.so.0 -> libspng.so.0.7.0
   ├── libspng.so.0.7.0
   └── pkgconfig
      └── spng.pc
```
From what I understand the meson build hooks run automatically. Not sure if this is a debug build or a release build. Also unsure if the checkPhase runs by default or not in this case

Reply:
I was checking in libpng, I think you also have to define manually the outputs
```
  outputs = [ "out" "dev" ];
  outputBin = "dev";
```

I think the include stuff is supposed to be part of dev

the build runs in two ways, on the build thing it talks about a cmake way and a meson way. If you remove the meson dependency, does it still build ?

If you run the build with `nix build .#libspng -L -vv` it should give you more verbose output and you should see if the checkphase is running.

Reply:
>  If you remove the meson dependency, does it still build ?

It doesn't build without without meson or ninja

@happysalada Did a few changes. PTAL

Reply:
Great!

does it produce two outputs result and result.dev
is the include directory correctly part of the result.dev ?
What is the output of the result ?

Reply:
The checkPhase is still not running as far as I can tell

Reply:
Does it even say anything in the logs ?
It should say that it's not checking.
Otherwise, you can try the `doCheck = true;`
If the checkPhase doesn't run then, it should say at least why in the logs.